### PR TITLE
feat(Table): add Table menu capability for each row

### PR DESCRIPTION
**Pull request by YossiSaadi**
The way to use the new API is declare a `rowId` prop for the menu component, which should the same as `id` passed to the row component:
```typescript
<TableRow id={id}>
  <TableRowMenu rowId={id}>
    <MenuItem title={"Item 1"} />
    <MenuItem title={"Item 2"} />
	// ...
  </TableRowMenu>
  <TableCell>{cellValue1}</TableCell>
  <TableCell>{cellValue2}</TableCell>
</TableRow>
```

also, you should now wrap your table with `<TableContainer>` - let's think of if there's a way to prevent it somehow
example:
```typescript
<TableContainer>
  <Table {...props} >
  // ...
  </Table>
</TableContainer>
```
https://monday.monday.com/boards/3532714909/pulses/6717924631

https://github.com/mondaycom/vibe/assets/30905362/8dcf8c1c-72c3-49e0-a0c2-93c81bfd3fdf

### CODE REVIEW

File: packages/core/src/tests/constants.ts [hunk 1]
```diff
@@ -110,12 +110,14 @@ export enum ComponentDefaultTestId {
...
   TABLE = "table",
+  TABLE_CONTAINER = "table-container",
   TABLE_BODY = "table-body",
   TABLE_VIRTUALIZED_BODY = "table-virtualized-body",
   TABLE_CELL = "table-cell",
   TABLE_HEADER = "table-header",
   TABLE_HEADER_CELL = "table-header-cell",
-  TABLE_ROW = "table-row"
+  TABLE_ROW = "table-row",
+  TABLE_MENU = "table-menu"
```
Comment (YossiSaadi) [inline packages/core/src/tests/constants.ts:120]: change to table-row-menu

File: packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  const resetTimeout = useCallback(() => {
+    clearTimeout(rowMouseLeaveTimeoutId.current);
+    rowMouseLeaveTimeoutId.current = null;
+  }, []);
+
+  const startTimeout = useCallback((callback: () => void) => {
+    rowMouseLeaveTimeoutId.current = setTimeout(() => {
+      callback();
+    }, 200);
```
Comment (YossiSaadi) [inline packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx:24]: 400?

File: packages/core/src/components/Table/TableRowMenu/TableRowMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      setTableMenuHidden,
+      setTableMenuShown,
+      menuButtonPosition,
+      onMouseOverRowMenu,
+      onMouseLeaveRowMenu
+    } = useTableRowMenu();
+
+    const onMenuHide = useCallback(() => {
+      setTableMenuHidden();
+      componentRef.current.blur();
```
Comment (YossiSaadi) [inline packages/core/src/components/Table/TableRowMenu/TableRowMenu.tsx:37]: check if not needed anymore

**Review by talkor — APPROVED** [review]
👏🏼 👏🏼 👏🏼

**Pull request merged**